src/layout/workspace.rs

```diff
@@ -426,13 +424,69 @@ impl<W: LayoutElement> Workspace<W> {
         ));
     }
 
+    fn animate_view_offset_to_column(&mut self, current_x: i32, idx: usize) {
+        let new_view_offset = self.compute_new_view_offset_for_column(current_x, idx);
+        self.animate_view_offset(current_x, idx, new_view_offset);
+    }
+
+    fn animate_view_offset_to_column_centered(&mut self, current_x: i32, idx: usize) {
+        if self.columns.is_empty() {
+            return;
+        }
+
+        let col = &self.columns[idx];
+        if col.is_fullscreen {
+            self.animate_view_offset_to_column(current_x, idx);
+            return;
+        }
+
+        let width = col.width();
+
+        // If the column is wider than the working area, then on commit it will be shifted to left
+        // edge alignment by the usual positioning code, so there's no use in trying to center it here.
+        if self.working_area.size.w <= width {
+            self.animate_view_offset_to_column(current_x, idx);
+            return;
+        }
+
+        let new_view_offset = -(self.working_area.size.w - width) / 2 - self.working_area.loc.x;
+
+        self.animate_view_offset(current_x, idx, new_view_offset);
+    }
+
     fn activate_column(&mut self, idx: usize) {
         if self.active_column_idx == idx {
             return;
         }
 
         let current_x = self.view_pos();
-        self.animate_view_offset_to_column(current_x, idx);
+        match self.options.center_column_on_focus {
+            CenterColumnOnFocus::Always => {
+                self.animate_view_offset_to_column_centered(current_x, idx)
+            }
+            CenterColumnOnFocus::NoFit => {
+                let source_x = self.column_x(self.active_column_idx);
+                let source_width = self.columns[self.active_column_idx].width();
```

Comment:
In case the focus jumps over multiple columns, do we perhaps want to compare against the neighbor column in the direction, rather than the source column? I.e. `idx + 1` if `active_column_idx > idx` and `idx - 1` otherwise?